artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/cursor/impl/PageSubscriptionImpl.java

```diff
@@ -1309,4 +1336,68 @@ public void remove() {
       public void close() {
       }
    }
+
+
+   private static class PagePositionSize {
```

Comment:
Why not just extend PagePosition (saves on objects)

Reply:
The reason I didn't extend is because PagePosition has methods like nextMessage() which increment the message number, etc.  If that is called then the stored size doesn't really make sense anymore as it won't match which is why I just created a new object.

Reply:
But this new object still holds PagePosition so its not been removed.

Reply:
I will fix this and extend PagePosition as if the position is incremented that is fine as the size will just go back to 0 unless set.